ajax.php

```diff
@@ -251,6 +251,33 @@
         $file->printResult();
         break;
 
+    case 'logxapievent':
+        // Trigger a Moodle log event for each xAPI statement
+        // that is dispatched by the H5P (hvp) object.
+        $hvpid = optional_param('hvpid', null, PARAM_INT);
+        $courseid = optional_param('courseid', null, PARAM_INT);
+        $jsonxapistatement = optional_param('xapistatement', null, PARAM_RAW);
+        $xapistatement = json_decode($jsonxapistatement, true);
+
+        $context = \context_module::instance($hvpid);
+
+        $event = \mod_hvp\event\hvp_xapi::create(array(
+            'objectid' => $hvpid,
+            'context' => $context,
+            'other' => array(
+                'actor' => $xapistatement['data']['statement']['actor'],
+                'verb' => $xapistatement['data']['statement']['verb'],
+                'object' => $xapistatement['data']['statement']['object'],
+                'context' => $xapistatement['data']['statement']['context'],
+                'result' => $xapistatement['data']['statement']['result']),
```

Comment:
Result is not always part of an xAPI event, for instance 'interacted' verb does not provide a result. Getting undefined php error.